app/models/supermarket.rb

```diff
@@ -1,3 +1,4 @@
 class Supermarket < ApplicationRecord
   has_many :items
-end
\ No newline at end of file
+  has_many :customers, through: :items
```

Comment:
Excellent use of an additional association to be able to more easily access customers